### DISCUSSION TITLE
Malloc branch

### DISCUSSION THREAD

**Discussion by krzysztofslusarski** [async-profiler/async-profiler#836] [General]
Hi, I wanted to compile async-profiler from malloc branch today, but I get `[WARN] Could not find Java_java_lang_ClassLoader_00024NativeLibrary_load` while starting the profiler. The lib is preloaded. @apangin do you have some working distribution from that branch ready?

**Comment by apangin** [comment]
No, that branch is obsoleted by an alternative solution I'm going to release soon, which does not require LD_PRELOAD.

**Reply by MattAlp** [reply]
Hello Andrei, I have skimmed the codebase and commits but do not see a reference to this solution being released- what is the state of "officially" tracing native memory today (mmap, malloc, mprotect...)?

**Reply by apangin** [reply]
This feature is pretty high in our plans. But so are other high priority features :)
Unfortunately, native memory profiling will not get in the next release, but will likely drop in the release after.
The rought timeline is several months. Please note this is just an estimation and subject to change. In no case this should be considered a committment.

**Reply by MattAlp** [reply]
Thank you! Is there anywhere where the unmerged / WIP code can be viewed? I am curious about the approach taken over LD_PRELOAD + would be happy to contribute if needed

**Reply by apangin** [reply]
This has not been published yet.
